src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,215 @@
+     * <Code>TaskList</Code>.
+     * @param input <Code>String</Code> format <Code>Task</Code> of to be
+     *              added to <Code>TaskList</Code>.
+     * @return      <Code>String</Code> output to be shown to user.
+     */
+    public String add(String input) {
+        String[] inputArr = input.split(" ");
+
+        Task newTask = null;
+        switch (inputArr[0]) {
+        case "deadline":
+            newTask = new Deadline(getDescription(input), getDate(input));
+            break;
+
+        case "event":
+            newTask = new Event(getDescription(input), getVenue(input));
+            break;
+
+        case "todo":
+            newTask = new ToDo(getDescription(input));
+            break;
+
+        default:
+            return "No task added.";
+        }
+
+        taskList.add(newTask);
+        int size = this.getSize();
+        return String.format("Got it. I've added this task:%n   %s%n"
+                + "Now you have %d task%s in the list.%n",
+                newTask, size, size == 1 ? "" : "s");
+    }
+
+
+    /**
+     * Deletes a <Code>Task</Code> from <Code>TaskList</Code>.
+     * @param input <Code>String</Code> user input command to delete
+     *              a <Code>Task</Code>.
+     * @return      <Code>String</Code> output to be shown to user.
+     */
+    public String delete(String input) {
+        int index = Integer.parseInt(input.split(" ")[1]) - 1;
+        String taskToDelete = taskList.get(index).toString();
+        taskList.remove(index);
+        int size = this.getSize();
+        return String.format("Got it. I've removed this task:%n   %s%n"
+                + "Now you have %d task%s in the list.%n",
+                taskToDelete, size, size == 1 ? "" : "s");
+    }
+
+    /**
+     * Overrides <Code>toString</Code> method of <Code>TaskList</Code> object.
+     * @return <Code>Task</Code> output to be shown to user.
+     */
+    @Override
+    public String toString() {
+        StringBuilder res = new StringBuilder(
+                String.format("Here are the tasks in your list:%n"));
+        for (int i = 0; i < taskList.toArray().length; i++) {
+            res.append(String.format("%d. %s%n", i + 1, taskList.get(i)));
+        }
+        return res.toString();
+    }
+
+    /**
+     * Marks <Code>Task</Code> with given index as done.
+     * @param index Index of task to be done. 1 based indexing.
+     * @return      <Code>String</Code> output to be shown to user.
+     */
+    public String markDone(int index) {
+        taskList.get(index - 1).markDone();
+        return String.format("Nice! I've marked this task as done:%n   %s%n",
+                taskList.get(index - 1));
+    }
+
+    /**
+     * Marks <Code>Task</Code> with given index as undone.
+     * @param index Index of task to be undone. 1 based indexing.
+     * @return      <Code>String</Code> output to be shown to user.
+     */
+    public String markUnDone(int index) {
```

Comment:
Should this be markUndone instead?

Reply:
Thank you for pointing that out! `markUnDone` has been renamed to `markUndone` in [commit 0a111e6](https://github.com/nus-cs2103-AY2223S1/ip/pull/15/commits/0a111e6647fcb5bf6706da0f8a986790952f6319).